costar_dataset/block_stacking_reader_torch.py

```diff
@@ -780,13 +917,32 @@ def from_standard_txt(cls, root, version, set_name, subset_name, split, feature_
             verbose=verbose, inference_mode=inference_mode, one_hot_encoding=one_hot_encoding,
             pose_name=pose_name,
             force_random_training_pose_augmentation=force_random_training_pose_augmentation,
-            single_batch_cube=single_batch_cube)
+            visual_mode=visual_mode,
+            single_batch_cube=single_batch_cube) for data_filename in data_filenames]
+        else:
+            data = cls(
+                data_filenames,
+                label_features_to_extract=label_features, data_features_to_extract=data_features,
+                total_actions_available=total_actions_available,
+                seed=seed, random_state=random_state,
+                is_training=is_training, random_augmentation=random_augmentation,
+                random_shift=random_shift,
+                output_shape=output_shape,
+                blend_previous_goal_images=blend_previous_goal_images,
+                num_images_per_example=num_images_per_example,
+                verbose=verbose, inference_mode=inference_mode, one_hot_encoding=one_hot_encoding,
+                pose_name=pose_name,
+                force_random_training_pose_augmentation=force_random_training_pose_augmentation,
+                visual_mode=visual_mode,
+                single_batch_cube=single_batch_cube)
 
         return data
 
     def __len__(self):
         """Return the lenth of file names
         """
+        if self.visual_mode:
+            return len(self.dataset[0])
```

Comment:
why only entry 0? I don't understand this

Reply:
This mode was to load the whole stack of  frames and joint vectors and then create embeddings for each video. But I changed the way it loads data now. Let me know if there is still a confusion after the next commit.
The goal is to have length be equal to the number of frames of the video